### ISSUE TITLE
unable to implement redo ("shift" "ctrl" "z") shortcut

### ISSUE THREAD

**Issue by njskalski** [gyscos/Cursive#240]
Hi,

in  Event:: there is no ShiftCtrlChar(). Can it be added? If not, then how to do a walkaround for linux-only?

Kindest Regards
Andrzej Skalski

**Comment by gyscos** [comment]
Hi!

First, on the ncurses backend, <kbd>Ctrl</kbd>+<kbd>Z</kbd> (and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Z</kbd>) will pause the application and return to the shell.
Second, there is no `ShiftCtrlChar` because it would be `Ctrl('Z')` instead (with a upper-case `Z`).
Third, doing this will not work because the terminal actually sends the same code if you press <kbd>Ctrl</kbd>+<kbd>Z</kbd> or <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Z</kbd> (usually it's the byte `0x18`): from the application point of view, there is no difference.

This shows the limitations of terminal input: many key combinations don't work, and there's nothing we can do from this application (that I know of).

You can try the `key_codes` example from the Cursive directory, it prints received events on the screen. This can help you debug what cursive receives when you press various key combinations.

**Comment by njskalski** [comment]
Right, thanks. I got a mind eclipse I guess. You are obviously right.

**Comment by njskalski** [comment]
Btw, just trying to disable system hooks for ctrl-c, ctrl-z etc. and don't know how to do it. Can you elaborate? Can I grab (somehow) termion backend and paly with it?

**Comment by gyscos** [comment]
* To try the termion backend, just compile cursive with the `termion-backend` feature. So to run the `key_codes` example:

```
cargo run --features termion-backend --example key_codes
```

* To catch these events in the ncurses backend as well, I'll investigate our options. I may try raw mode, or maybe just catch `SIGINT` signal and send <kbd>Ctrl</kbd>+<kbd>C</kbd> from there. Not sure if I can catch `SIGTSTP` for <kbd>Ctrl</kbd>+<kbd>Z</kbd>.

**Comment by njskalski** [comment]
Tried termion the command you gave, and ctrl-c still exits program.

**Comment by gyscos** [comment]
Ah yes, I bound the Ctrl-C event sent by termion to the Exit event which
stops the clevent loop. That could be configurable if I can get it to work
on the other backends.

**Comment by njskalski** [comment]
Oh. Ok. So let me know if you decide to proceed with that.
